dkroutingtool/src/py/optimization.py

```diff
@@ -1454,7 +1508,46 @@ def segment_unload_route(route):
                     del routes_for_mapping[zone_route]
                     del vehicles[zone_route]
                     solution.zone_route_map[zone_to_segment] = zone_routes
-                    
+
+    if color_naming:
+        new_routes_for_mapping = {}
+        new_vehicles = {}
+        for key in routes_for_mapping:
```

Comment:
Could we put this in a function like: new_routes_for_mappings, new_vehicles = reconcile_route_keynames(routes_for_mapping)